src/components/Faucet/Stripe.tsx

```diff
@@ -0,0 +1,227 @@
+import { Box, Button, Flex, Link, Spinner, Text, useToast } from '@chakra-ui/react'
+import { EmbeddedCheckout, EmbeddedCheckoutProvider } from '@stripe/react-stripe-js'
+import { loadStripe } from '@stripe/stripe-js'
+import { errorToString, useClient } from '@vocdoni/react-providers'
+import { useCallback, useEffect, useState } from 'react'
+import { Trans, useTranslation } from 'react-i18next'
+import { Navigate } from 'react-router-dom'
+import { useAccount } from 'wagmi'
+
+const STRIPE_PUBLIC_KEY = import.meta.env.STRIPE_PUBLIC_KEY
+const stripePromise = loadStripe(STRIPE_PUBLIC_KEY)
+
+type CheckoutFormProps = {
+  amount?: string
+  returnURL?: string
+}
+
+type StatusData = {
+  status: string | null
+  customer_email: string | null
+  faucet_package: string | undefined
+  recipient: string | null
+  quantity: number | null
+}
+
+export const CheckoutForm = ({ amount, returnURL }: CheckoutFormProps) => {
+  const { address } = useAccount()
+  const { client } = useClient()
+  const origin = window.location.origin
+
+  const fetchClientSecret = useCallback(async () => {
+    if (address) {
+      let uri = `${client.faucetService.url}/createCheckoutSession/${address}`
+      if (amount && !isNaN(parseInt(amount))) {
+        uri = `${uri}/${amount}`
+      }
```

Comment:
Always specify the radix
```suggestion
      if (amount && !isNaN(parseInt(amount, 10))) {
        uri = `${uri}/${amount}`
      }
```